### DISCUSSION TITLE
Step function in conversations not working in webhook mode

### DISCUSSION THREAD

**Discussion by festars** [nutgram/nutgram#492] [General]
``use SergiX44\Nutgram\Conversations\Conversation;
use SergiX44\Nutgram\Nutgram;

class MyConversation extends Conversation {

    public function start(Nutgram $bot)
    {
        $bot->sendMessage('This is the first step!');
        $this->next('secondStep');
    }

    public function secondStep(Nutgram $bot)
    {
        $bot->sendMessage('Bye!');
        $this->end();
    }
}

$bot = new Nutgram($_ENV['TOKEN']);

$bot->onCommand('start', MyConversation::class);

$bot->run();

**Comment by festars** [comment]
Kindly note in Polling mode its works without issues in webhook i only get 'This is the first step!' only

**Comment by Lukasss93** [comment]
Have you seen this page: https://nutgram.dev/docs/configuration/cache ?
You need a persistent cache adapter to use "conversations" feature: https://nutgram.dev/docs/usage/conversations

**Reply by festars** [reply]
Great, it has worked
